packages/core/echo/echo-db/src/core-db/automerge-repo-replacement/repo.ts

```diff
@@ -0,0 +1,194 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { DeferredTask, sleep } from '@dxos/async';
+import {
+  type DocumentId,
+  generateAutomergeUrl,
+  parseAutomergeUrl,
+  interpretAsDocumentId,
+  type AnyDocumentId,
+} from '@dxos/automerge/automerge-repo';
+import { type Stream } from '@dxos/codec-protobuf';
+import { Resource } from '@dxos/context';
+import { invariant } from '@dxos/invariant';
+import { PublicKey } from '@dxos/keys';
+import { log } from '@dxos/log';
+import {
+  type DataService,
+  type BatchedDocumentUpdates,
+  type DocumentUpdate,
+} from '@dxos/protocols/proto/dxos/echo/service';
+
+import { DocHandleReplacement } from './doc-handle';
+
+const UPDATE_BATCH_INTERVAL = 100;
+
+export class RepoReplacement extends Resource {
```

Comment:
as an idea: add flush method which exectes write job immediatelly (no debounce) and use it in db.flush() -- you won't have to compare heads that way